crates/biome_js_analyze/src/analyzers/nursery/no_duplicate_test_hooks.rs

```diff
@@ -0,0 +1,221 @@
+                                    };
+                                };
+                            })
+                    }
+                }
+            }
+            WalkEvent::Leave(node) => {
+                // When the visitor exits a function, if it matches the node of the top-most
+                // entry of the stack and the `has_yield` flag is `false`, emit a query match
+                if let Some(node) = JsCallExpression::cast_ref(node) {
+                    if let Ok(callee) = node.callee() {
+                        if callee.contains_a_test_pattern().is_ok() {
+                            if let Some(function_name) = get_function_name(&callee) {
+                                if function_name.text_trimmed() == "describe" {
+                                    self.stack.pop();
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+    }
+}
+
+// Declare a query match struct type containing a JavaScript function node
+pub(crate) struct DuplicateHooks(JsCallExpression);
+
+impl QueryMatch for DuplicateHooks {
+    fn text_range(&self) -> TextRange {
+        self.0.range()
+    }
+}
+
+impl Queryable for DuplicateHooks {
+    // `Input` is the type that `ctx.match_query()` is called with in the visitor
+    type Input = Self;
+    type Language = JsLanguage;
+    // `Output` if the type that `ctx.query()` will return in the rule
+    type Output = JsCallExpression;
+    type Services = ();
+
+    fn build_visitor(
+        analyzer: &mut impl AddVisitor<Self::Language>,
+        _: &<Self::Language as Language>::Root,
+    ) {
+        // Register our custom visitor to run in the `Syntax` phase
+        analyzer.add_visitor(Phases::Syntax, DuplicateHooksVisitor::default);
+    }
+
+    // Extract the output object from the input type
+    fn unwrap_match(_: &ServiceBag, query: &Self::Input) -> Self::Output {
+        query.0.clone()
+    }
+}
+
+impl Rule for NoDuplicateTestHooks {
+    type Query = DuplicateHooks;
+    type State = ();
+    type Signals = Option<Self::State>;
+    type Options = ();
+
+    fn run(_: &RuleContext<Self>) -> Self::Signals {
+        Some(())
+    }
+
+    fn diagnostic(ctx: &RuleContext<Self>, _: &Self::State) -> Option<RuleDiagnostic> {
+        let node = ctx.query();
+        let callee = node.callee().ok()?;
+        let node_name = get_function_name(&callee)?;
+        Some(
+            RuleDiagnostic::new(
+                rule_category!(),
+                ctx.query().range(),
+                markup! {
+                    "Disallow duplicate setup and teardown hooks."
+                },
+            )
+            .note(markup! {
+                "Disallow "<Emphasis>{node_name.text_trimmed()}</Emphasis>" duplicacy inside the describe function."
+            }),
+        )
+    }
+}
+
+fn get_function_name(callee: &AnyJsExpression) -> Option<JsSyntaxToken> {
```

Comment:
We already have `get_function_name` [here](https://github.com/vasucp1207/biome/blob/09724201775020857b938cd9e41d50093384794c/crates/biome_js_analyze/src/analyzers/nursery/no_skipped_tests.rs#L55). 

Maybe we can move inside `expr_ext.rs`, this it's more generic now

Reply:
These both are a little different, here it returns the `object` part of callee not the `member`. Maybe we can move these after seeing how other rules require them? For now, let it would be here.